packages/from-hardhat/package.json

```diff
@@ -0,0 +1,52 @@
+{
+  "name": "@truffle/from-hardhat",
+  "description": "Import Hardhat project information into Truffle-native formats",
+  "license": "MIT",
+  "author": "g. nicholas d'andrea <gnidan@trufflesuite.com>",
+  "homepage": "https://github.com/trufflesuite/truffle/tree/master/packages/from-hardhat#readme",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/trufflesuite/truffle.git",
+    "directory": "packages/from-hardhat"
+  },
+  "bugs": {
+    "url": "https://github.com/trufflesuite/truffle/issues"
+  },
+  "version": "0.1.0-0",
+  "main": "dist/src/index.js",
+  "scripts": {
+    "build": "ttsc",
+    "prepare": "yarn build",
+    "test": "exit 0"
+  },
+  "dependencies": {
+    "@truffle/compile-common": "^0.7.32",
+    "@truffle/compile-solidity": "^6.0.38",
+    "@truffle/config": "^1.3.34",
+    "debug": "^4.3.1",
+    "find-up": "^2.1.0"
+  },
+  "devDependencies": {
+    "@types/find-up": "^2.1.0",
+    "@types/node": "^18.6.5",
+    "hardhat": "^2.10.1",
```

Comment:
@cds-amal @fainashalts FYI I realized that we can benefit from Hardhat's own types by keeping this as a devDependency safely... if we just don't actually export anything with these types in the signature. This is the approach I'm taking here.

Just letting you know because this breaks the assumptions of our earlier conversation.

Reply:
how does this compare to making it an optional peerDepenency?

Reply:
> how does this compare to making it an optional peerDepenency?

Good question! Firstly, it's worth noting that [`package.json`'s `peerDependenciesMeta` field](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta) (which is how you mark peerDeps as optional) was added in NPM v7, and we need to support v6 still. But this is a minor quibble.

The main concern here, IMO, is the question "what does Truffle's Node.js-land need from the `"hardhat"` NPM package?"
- Types means _at least_ a `devDep`
- We need more than `devDep` if-and-only-if we want to have a shared Node.js runtime between Truffle and Hardhat... as in, iff we want to use Hardhat as a library

I looked at using HH as a library, but, like Truffle, it's really not intended for use via straight `require()` (or `import`). I got it working in #5410... so this approach is viable albeit awkward. But think about what we're saying when Truffle does `require("hardhat")`... like, _"here you go, Node! have another, different one of **me**!"_. Even without the risk of cross-clobbering each other's globals / other runtime modification shenanigans, the memory footprint and whatnot... oy!

So my conclusion was basically... screw `peerDependencies`, HH is prescriptive: you use it via `npx hardhat`. Truffle can use HH via `npx hardhat`. Keep the playgrounds separate :)

Reply:
Probably worth following up that there is a counter-argument that we can just use `peerDependencies` for the `$(npm bin)` or whatever. But still... I think there's value in treating HH as an executable... like, you're not going to try to shove Docker into your `package.json` - you just shell out to `docker` and error nicely if it's not there.

Reply:
Ah that makes sense, and feel free to ignore my repeat question in the review below.

I was just thinking that it'd be very nice if there was some way to import _just_ the types from the hardhat package without needing to add all of their transitive dependencies into our `node_modules` tree and our `yarn.lock` (even if it is just for dev purposes).

If our usage of their types is minimal, perhaps it'd be worth extracting _only_ the types that we need into a local type declaration? Though I could also easily argue against that approach due to the potential effort involved in doing that, along with the increased maintenance burden that will be required.

Sure would be nice if there was a code-gen utility for that sort of thing, though!

Reply:
Hm, I see your concern. Not sure copy+pasting the types is acceptable? Sounds like follow-on work? Or maybe we could get HH to publish types separately?

Reply:
Yeah, I think _if_ we make changes to this approach, it's definitely something that's a future concern.

Reply:
Ah, I resolved this, but realized that neither of the two people who you pinged at the start have replied, so I've unresolved it.

Reply:
Otherwise please consider my part in this resolved.